backend/src/router/block/get-hash-by-id.ts

```diff
@@ -13,7 +13,7 @@ export const router = trpc.router<Context>().query("getHashById", {
     let selection = indexerDatabase
       .selectFrom("blocks")
       .select("block_hash as hash");
-    if (isNaN(parseInt(blockId))) {
```

Comment:
If we take blockHash from issue (`4usxPdL68XbQXBeZPGj56wH1FAqrXhuFygHTz27xodMU`) and `console.log("4usxPdL68XbQXBeZPGj56wH1FAqrXhuFygHTz27xodMU")` we'll see output `4` and `isNaN(parseInt("4usxPdL68XbQXBeZPGj56wH1FAqrXhuFygHTz27xodMU"))` returns `false`. If we use `Number` instead of `parseInt` - works as expected.

Reply:
This still looks incorrect to me. It is possible that block hash will only contain digits, so it will appear as a very big number. Let’s change this condition to just checking the length (if it is 43-44 characters long, search it as block hash, otherwise, check if it is a number and search as if it is block height, and otherwise skip search completely, as it does not make any sense to search lets say frol.near block)